sdk/src/main/java/com/amplitude/experiment/FlagApi.kt

```diff
@@ -60,12 +64,28 @@ internal class SdkFlagApi(
             override fun onResponse(call: Call, response: Response) {
                 try {
                     Logger.d("Received fetch flags response: $response")
-                    val body = response.body?.string() ?: ""
-                    val flags = json.decodeFromString<List<EvaluationFlag>>(body)
-                        .associateBy { it.key }
-                    future.complete(flags)
+
+                    if (response.isSuccessful) {
+                        val body = response.body?.string() ?: ""
+
+                        // Check if the response is a valid JSON array
+                        if (body.startsWith("[") && body.endsWith("]")) {
```

Comment:
this seems unnecessary. 

The issue lists the Exception is a JsonDecodingException which is a subclass of IllegalArgumentException, meaning the exception was not caught. 

I.e. the new Serialization exception catch block is sufficient to handle bad input.

Reply:
So I would remove this if else statement, and let the decodeFromString call throw, to be caught by the SerializationException catch block.